### PR TITLE
Split the locator class name getting in a separate method.

### PR DESCRIPTION
**Pull request by burzum**

### CODE REVIEW

File: src/ORM/Locator/TableLocator.php [hunk 1]
```diff
@@ -146,10 +146,7 @@ public function get($alias, array $options = [])
             $options += $this->_config[$alias];
         }
 
-        if (empty($options['className'])) {
-            $options['className'] = Inflector::camelize($alias);
-        }
-        $className = App::className($options['className'], 'Model/Table', 'Table');
+        $className = $this->_getClassName($alias, $options);
```
Comment (lorenzo) [inline src/ORM/Locator/TableLocator.php]: Should be `$className = $options['className'] = $this->_getClassName($alias, $options);`

**Comment by dereuromark** [comment]
:+1:

**Comment by markstory** [comment]
A few [tests failed](https://travis-ci.org/cakephp/cakephp/jobs/92019737).

**Comment by burzum** [comment]
@markstory no idea why they fail:

```sh
C:\xampp\htdocs\crm\vendor\cakephp\cakephp>vendor\bin\phpunit tests\TestCase\ORM\TableRegistryTest.php
PHPUnit 4.4.1 by Sebastian Bergmann.

Configuration read from C:\xampp\htdocs\crm\vendor\cakephp\cakephp\phpunit.xml.dist

.......

Time: 115 ms, Memory: 5.75Mb

OK (7 tests, 8 assertions)`
```

**Comment by markstory** [comment]
It looks like tests outside the unit tests for table registry are the troublesome ones.

**Comment by burzum** [comment]
@markstory I can't run all tests without trouble for some month. It keeps breaking when it tries to run the AuthComponent tests. I haven't had the time and mood to figure out the cause.

```sh
Configuration read from C:\xampp\htdocs\wa-crm\vendor\cakephp\cakephp\phpunit.xml.dist

.............................................................   61 / 6186 (  0%)
................................SSSSSSSSSSS...............S..  122 / 6186 (  1%)
....SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS  183 / 6186 (  2%)
SSSSSSSSSS...................................................  244 / 6186 (  3%)
.............................................................  305 / 6186 (  4%)
.............................................................  366 / 6186 (  5%)
.........................................................S...  427 / 6186 (  6%)
..............................................PHP Fatal error:  Call to undefined function Cake\Controller\Component\__d() in C:\xampp\htdocs\wa-
crm\vendor\cakephp\cakephp\src\Controller\Component\AuthComponent.php on line 454
PHP Stack trace:
PHP   1. {main}() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\phpunit:0
PHP   2. PHPUnit_TextUI_Command::main() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\phpunit:62
PHP   3. PHPUnit_TextUI_Command->run() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\TextUI\Command.php:138
PHP   4. PHPUnit_TextUI_TestRunner->doRun() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\TextUI\Command.php:186
PHP   5. PHPUnit_Framework_TestSuite->run() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\TextUI\TestRunner.php:423
PHP   6. PHPUnit_Framework_TestSuite->run() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framework\TestSuite.php:751
PHP   7. PHPUnit_Framework_TestSuite->run() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framework\TestSuite.php:751
PHP   8. PHPUnit_Framework_TestCase->run() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framework\TestSuite.php:751
PHP   9. PHPUnit_Framework_TestResult->run() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framework\TestCase.php:722
PHP  10. PHPUnit_Framework_TestCase->runBare() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framework\TestResult.php:
643
PHP  11. PHPUnit_Framework_TestCase->runTest() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framework\TestCase.php:76
6
PHP  12. ReflectionMethod->invokeArgs() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framework\TestCase.php:881
PHP  13. Cake\Test\TestCase\Controller\Component\AuthComponentTest->testIdentify() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\p
hpunit\src\Framework\TestCase.php:881
PHP  14. Cake\Controller\Component\AuthComponent->identify() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\tests\TestCase\Controller\Component\Au
thComponentTest.php:163
PHP  15. Cake\Controller\Component\AuthComponent->_setDefaults() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\src\Controller\Component\AuthCompo
nent.php:758

Fatal error: Call to undefined function Cake\Controller\Component\__d() in C:\xampp\htdocs\crm\vendor\cakephp\cakephp\src\Controller\Component
\AuthComponent.php on line 454

Call Stack:
    0.0002     128856   1. {main}() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\phpunit:0
    0.0047     532456   2. PHPUnit_TextUI_Command::main() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\phpunit:62
    0.0047     536336   3. PHPUnit_TextUI_Command->run() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\TextUI\Command.
php:138
    2.2218   62670224   4. PHPUnit_TextUI_TestRunner->doRun() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\TextUI\Com
mand.php:186
    2.2266   62993856   5. PHPUnit_Framework_TestSuite->run() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\TextUI\Tes
tRunner.php:423
    2.4143   63005848   6. PHPUnit_Framework_TestSuite->run() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framework\
TestSuite.php:751
   14.0810   77288896   7. PHPUnit_Framework_TestSuite->run() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framework\
TestSuite.php:751
   14.2493   77826368   8. PHPUnit_Framework_TestCase->run() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framework\T
estSuite.php:751
   14.2493   77826336   9. PHPUnit_Framework_TestResult->run() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framework
\TestCase.php:722
   14.2580   77919760  10. PHPUnit_Framework_TestCase->runBare() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framewo
rk\TestResult.php:643
   14.3241   78009536  11. PHPUnit_Framework_TestCase->runTest() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framewo
rk\TestCase.php:766
   14.3241   78010104  12. ReflectionMethod->invokeArgs() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\vendor\phpunit\phpunit\src\Framework\Test
Case.php:881
   14.3241   78010328  13. Cake\Test\TestCase\Controller\Component\AuthComponentTest->testIdentify() C:\xampp\htdocs\crm\vendor\cakephp\cakeph
p\vendor\phpunit\phpunit\src\Framework\TestCase.php:881
   14.3254   78030808  14. Cake\Controller\Component\AuthComponent->identify() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\tests\TestCase\Contr
oller\Component\AuthComponentTest.php:163
   14.3254   78030856  15. Cake\Controller\Component\AuthComponent->_setDefaults() C:\xampp\htdocs\crm\vendor\cakephp\cakephp\src\Controller\C
omponent\AuthComponent.php:758

C:\xampp\htdocs\crm\vendor\cakephp\cakephp [locator +6 ~0 -0 !]
```

**Comment by lorenzo** [comment]
@burzum how did you run the tests?

**Comment by burzum** [comment]
@lorenzo from the CakePHP folder:

```sh
composer install
vendor\bin\phpunit
```

Something wrong with that?

**Comment by AD7six** [comment]
@burzum there is a finite list of errors from the build

> There were 4 errors:
> 1) Cake\Test\TestCase\ORM\QueryRegressionTest::testComplexNestedTypesInJoinedWhere
> InvalidArgumentException: Cannot convert value to string

These are tests which make use of the methods you've modified.

**Comment by lorenzo** [comment]
@burzum I don't see a problem with that... but it is quite weird. Ha you tried running the tests by putting cakephp outside the vendors folder?

File: src/ORM/Locator/TableLocator.php [hunk 2]
```diff
@@ -176,6 +176,21 @@ public function get($alias, array $options = [])
...
+     *
+     * @param string $alias The alias name you want to get.
+     * @param array $options Table options array.
+     * @return string
+     */
+    protected function _getClassName($alias, array $options = [])
+    {
+        if (empty($options['className'])) {
+            $options['className'] = Inflector::camelize($alias);
+        }
```
Comment (markstory) [inline src/ORM/Locator/TableLocator.php]: Shouldn't the camelize() on line 148-150 be removed?

**Comment by lorenzo** [comment]
I'll remove the extra code

**Comment by lorenzo** [comment]
Thanks @burzum !

**Pull request merged**